### ISSUE TITLE
Signal crashes on launch

### ISSUE THREAD

**Issue by nicholasmullikin** [signalapp/Signal-Android#13115]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal crashes about 0.5 seconds after launch.
```
08-16 22:53:57.601 31576 31622 W ContactRecordProcessor: Newly discovering a registered user via storage service. Saving a storageId for them.
08-16 22:53:57.606 31576 31622 I DefaultStorageRecordPro: [0][SignalContactRecord] [Remote Update] [ID, ProfileKey]
08-16 22:53:57.606 31576 31622 I DefaultStorageRecordPro: [0][SignalContactRecord] [Local Update] [ID, E164, UnregisteredTimestamp]
08-16 22:53:57.608 31576 31622 W RecipientTable: [applyStorageSyncContactUpdate] Failed to update a user by storageId.
08-16 22:53:57.608 31576 31622 W RecipientTable: [applyStorageSyncContactUpdate] Found user RecipientId::637. Possibly merging.
08-16 22:53:57.614 31120 31133 V ClearcutMetricXmitter: Transmission is done.
08-16 22:53:57.621 31576 31622 E BaseJob : Encountered a fatal exception. Crash imminent.
08-16 22:53:57.621 31576 31622 E BaseJob : java.lang.IllegalStateException: Should not occur in a non-PNP world!
08-16 22:53:57.621 31576 31622 E BaseJob : 	at org.thoughtcrime.securesms.database.MessageTable.insertSessionSwitchoverEvent(MessageTable.kt:1300)
```
Following log taken via `adb logcat`

### Steps to reproduce
- Tap on Signal from homescreen

**Actual result:** Signal crashes
**Expected result:** Signal opens

### Device info
**Device:** Pixel 5a
**Android version:** 13
**Signal version:** 6.28.6 & 6.29.2

### Link to debug log
Contains the app crashing multiple times:
[SignalCrashLog.txt](https://github.com/signalapp/Signal-Android/files/12365063/SignalCrashLog.txt)

**Comment by greyson-signal** [comment]
Hi there, I know this is a long-shot, but is there any chance you previously had a custom build to mess around with phone number privacy or usernames early?

**Comment by nicholasmullikin** [comment]
Hmm I don't think so, but I'm not 100% sure. I will note that recently one of my contacts recently created a new signal account where I accepted the message request, but on signal desktop, it still says: `This person isn’t using Signal, Signal Desktop does not support messaging non-Signal contacts. Ask this person to install Signal for a more secure messaging experience.` even though I'm receiving messages.

**Comment by greyson-signal** [comment]
Ah, that's interesting about the contact. I do see evidence in the logs that the contact that is causing the crash does have data related to changing their registered status.

> Hmm I don't think so, but I'm not 100% sure.

Do you know if you were building your own version of Signal, or if you were using a non-official Signal build? Are you using an official copy from the Play Store or our website right now? If you are using an official copy right now, did you ever import a backup from a non-official build?

**Comment by nicholasmullikin** [comment]
I don't think I ever built my own version of Signal. I think the furthest I would have gone was installing the beta from time to time

**Comment by aarondemolder** [comment]
Also experiencing this issue of instant crashing on app launch

Also added a new contact yesterday which needed a contact merge and took a while to sync up properly, unsure if that's coincidence

Pixel 4, Android 13, 6.28.6, Play store (non-beta, and never been signed up or used unofficial builds), also just tried beta 6.30.1, still crashing

**Comment by greyson-signal** [comment]
Thank you for the new logs! I put extra debugging information in 6.30.1 which has helped me figure out the issue. A fix will go out in 6.30.2.
